src/databricks/labs/ucx/source_code/notebooks/cells.py

```diff
@@ -183,16 +188,37 @@ def language(self):
     def is_runnable(self) -> bool:
         return True  # TODO
 
+    @staticmethod
+    def _split(code) -> list[str]:
+        """Split pip cell code into multiple arguments
+
+        Note:
+            PipCell should be a pip command, i.e. single line possible spanning multilines escaped with backslashes \
+
+        Sources:
+            https://docs.databricks.com/en/libraries/notebooks-python-libraries.html#manage-libraries-with-pip-commands
+        """
+        match = re.search(r"(?<!\\)\n", code)
+        if match:
+            code = code[: match.start()]  # Remove code after non-escaped newline
+        code = code.replace("\\\n", " ")
+        lexer = shlex.split(code, posix=True)
+        return list(lexer)
+
     def build_dependency_graph(self, graph: DependencyGraph) -> list[DependencyProblem]:
-        # TODO: this is very basic code, we need to improve it
-        splits = re.split(r" |\n", self.original_code)
-        if len(splits) < 3:
-            return [DependencyProblem("library-install-failed", f"Missing arguments in '{self.original_code}'")]
-        if splits[1] != "install":
-            return [DependencyProblem("library-install-failed", f"Unsupported %pip command: {splits[1]}")]
-        # TODO: we need to support different formats of the library name and etc
-        library = splits[2]
-        return graph.register_library(library)
+        argv = self._split(self.original_code)
+        try:
+            cmd_name, cmd_args = parse_command(argv[1:])  # Skipping %pip
+        except PipError as e:
+            return [DependencyProblem("library-install-failed", str(e))]
+        if cmd_name != "install":
+            return [DependencyProblem("library-install-failed", f"Unsupported %pip command: {cmd_name}")]
+        if len(cmd_args) == 0:
+            return [DependencyProblem("library-install-failed", "Missing arguments in '%pip install'")]
+        install = InstallCommand(name="install", summary="Install packages.")
+        _, args = install.parse_args(cmd_args)
+        # Installation arguments are needed to pass installation flags
+        return graph.register_library(*args, installation_arguments=cmd_args if args != cmd_args else [])
```

Comment:
If a pip command has additional arguments defined besides the to be installed libraries, then these arguments are passed to the `register_library` so that it can be reused in the pip install